backend/agentd/agentd.go

```diff
@@ -141,8 +149,15 @@ func New(c Config, opts ...Option) (*Agentd, error) {
 	// functions, which prevent us from modifying the actual middleware logic at
 	// runtime, so we need this workaround
 	router := mux.NewRouter()
-	router.HandleFunc("/", a.webSocketHandler)
-	router.Use(authenticate, authorize, entityLimit, agentLimit)
+
+	healthRouter := routers.NewHealthRouter(
+		actions.NewHealthController(a.store, a.client.Cluster, a.etcdClientTLSConfig),
+	)
+	healthRouter.Mount(router)
+
+	route := router.NewRoute().Subrouter()
```

Comment:
Not sure I understand this line.

Reply:
This creates a new subrouter that apply the authentication & authorization middleware to.